### PR TITLE
VIITE-967 - check change information works also with roadaddress history

### PR DESCRIPTION
**Pull request by pedrosag**
Small adaptation to the apply_change_information_to_road_address_links in order to validate road addresses

### CODE REVIEW

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala [hunk 1]
```diff
@@ -298,6 +298,12 @@ object RoadAddressDAO {
     else
       ""
 
+    val valid = if(includeValid) {
+      "AND (ra.end_date is null OR ra.end_data >= sysdate) AND ra.terminated = 0 "
```
Comment (tvenhola) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala]: Should not include a date check: all road address history should be moved with vvh change table to the new link. Instead only termination should be checked.

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala [hunk 2]
```diff
@@ -280,7 +280,7 @@ object RoadAddressDAO {
     }
   }
 
-  def fetchByLinkId(linkIds: Set[Long], includeFloating: Boolean = false, includeHistory: Boolean = true): List[RoadAddress] = {
+  def fetchByLinkId(linkIds: Set[Long], includeFloating: Boolean = false, includeHistory: Boolean = true, includeValid: Boolean = false): List[RoadAddress] = {
```
Comment (tvenhola) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala]: Should not be called "valid" as it falsely points to valid_from and valid_to fields.
includeTerminated: Boolean = true

File: digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala (hunk 1 above)
Comment (tvenhola) [reply digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala]: use includeTerminated flag and change to 
if (!includeTerminated) {
  "AND ra.terminated = 0 "
...

Comment (Campanha) [inline digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/dao/RoadAddressDAO.scala]: "ra.end_data" instead "ra.end_date" should have done problems tho

**Pull request merged**